src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -77,8 +78,8 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
      */
     String w3Link = getLink(W3_CSE_ID, query);
```

Comment:
Quick note: To allow for this to easily loop over a list of clients, I think you'll want the CSE_IDs to be associated with the clients so you can do something like:

```
for (Client client : clients) {
  String link = getLink(client.getCseId(), query);
  Card card = client.search(link);
  allCards.add(card);
}
```

To associate the CSE ID with the client, I think you can either define them in the Clients themselves or if you want them all here for ease, you can pass it through in the constructor? So you would call `new W3SchoolClient(W3_CSE_ID)` and add a `getCseId()` method in your clients.